ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapCardAdapter.kt

```diff
@@ -22,32 +22,57 @@ import androidx.core.content.res.ResourcesCompat
 import androidx.recyclerview.widget.RecyclerView
 import com.google.android.ground.R
 import com.google.android.ground.databinding.LoiCardItemBinding
+import com.google.android.ground.databinding.SuggestLoiCardItemBinding
+import com.google.android.ground.model.job.Job
 import com.google.android.ground.model.locationofinterest.LocationOfInterest
-import com.google.android.ground.ui.home.mapcontainer.MapCardAdapter.ViewHolder
+import com.google.android.material.card.MaterialCardView
 
 /**
  * An implementation of [RecyclerView.Adapter] that associates [LocationOfInterest] data with the
  * [ViewHolder] views.
  */
-class MapCardAdapter : RecyclerView.Adapter<ViewHolder>() {
+class MapCardAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
 
   private var focusedIndex: Int = 0
+  private var indexOfLastLoi: Int = -1
   private val itemsList: MutableList<MapCardUiData> = mutableListOf()
   private lateinit var cardFocusedListener: (MapCardUiData?) -> Unit
   private lateinit var collectDataListener: (MapCardUiData) -> Unit
 
-  /** Creates a new [ViewHolder] item without any data. */
-  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
-    val binding = LoiCardItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
-    return ViewHolder(binding)
-  }
+  /** Creates a new [RecyclerView.ViewHolder] item without any data. */
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder =
+    if (viewType == R.layout.loi_card_item) {
+      LoiViewHolder(LoiCardItemBinding.inflate(LayoutInflater.from(parent.context), parent, false))
+    } else {
+      SuggestLoiViewHolder(
+        SuggestLoiCardItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+      )
+    }
 
-  /** Binds [LocationOfInterest] data to [ViewHolder]. */
-  override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+  override fun getItemViewType(position: Int): Int =
+    if (position <= indexOfLastLoi) {
```

Comment:
Is it possible to have more than one suggest loi type card?

Reply:
Yes

Reply:
So every element after the last loi card will be a suggest loi card